src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -231,22 +243,73 @@ private int getStatesPercentageDifference(double defaultVideoHeight, double reso
   }
 
   private <T> T usingVideoContext(final Function<WebElement, T> fun) {
-    return usingVideoIframeContext(webDriver -> fun.apply(driver.findElement(By.tagName("video"))));
+    return usingImaBridge(webDriver -> fun.apply(driver.findElement(By.tagName("video"))));
   }
 
-  private <T> T usingVideoIframeContext(final Function<WikiaWebDriver, T> fun) {
-    final WebElement iframe = driver.findElement(By.cssSelector(videoIframeSelector));
-    driver.switchTo().frame(iframe);
+  private <T> T usingImaBridge(final Function<WikiaWebDriver, T> fun) {
+    driver.switchTo().frame(driver.findElement(imaBridgeSelector));
     final T result = fun.apply(driver);
     driver.switchTo().defaultContent();
     return result;
   }
 
   private void usingAdFrame(Runnable f) {
-    final WebElement iframe = driver.findElement(By.cssSelector("#" + slot + " .provider-container iframe"));
-    driver.switchTo().frame(iframe);
+    wait.forElementPresent(adIframeSelector);
+    driver.switchTo().frame(driver.findElement(adIframeSelector));
     f.run();
     driver.switchTo().defaultContent();
   }
 
+  public boolean isPauseLayerVisible() {
+    wait.forElementVisible(pauseOverlaySelector);
+    return true;
+  }
+
+  public boolean isPauseLayerNotVisible() {
+    wait.forElementNotVisible(pauseOverlaySelector);
+    return true;
+  }
+
+  public void clickOnArea(int area) {
+    clickElementInsideAd(By.id("area" + area));
+  }
+
+  public void waitForVideoStart() {
+    isPauseLayerVisible();
+  }
+
+  public void waitForVideoPlayerHidden() {
+    isPauseLayerNotVisible();
+  }
+
+  public double getAdVideoHeight() {
+    return driver.findElement(pauseOverlaySelector).getSize().getHeight();
+  }
+
+  public Double getCurrentTime() {
+    String result;
+
+    if (hasMobileVideoElement()) {
+      result = usingVideoContext(video -> video.getAttribute("currentTime"));
+    } else {
+      result = driver.findElement(getMobileVideoSelector()).getAttribute("currentTime");
+    }
+    return Double.parseDouble(result);
+  }
+
+  private boolean hasMobileVideoElement() {
+    return driver.findElements(getMobileVideoSelector()).size() == 0;
```

Comment:
It isn't when reading the code. There should be `> 0` IMHO. @v3rth, czy to je*nie? :trollface: